src/core/mesh/qgsmeshlayer.cpp

```diff
@@ -73,6 +73,27 @@ void QgsMeshLayer::setDefaultRendererSettings()
   }
 }
 
+void QgsMeshLayer::createSimplifiedMeshes()
+{
+  if ( mSimplificationSettings.isEnabled() && !hasSimplifiedMeshes() )
+  {
+    double reductionFactor = mSimplificationSettings.reductionFactor();
+
+    QVector<QgsTriangularMesh *> simplifyMeshes =
+      mTriangularMeshes[0]->simplifyMesh( reductionFactor );
+
+    for ( int i = 0; i < simplifyMeshes.count() ; ++i )
+    {
+      mTriangularMeshes.emplace_back( simplifyMeshes[i] );
+    }
+  }
+}
+
+bool QgsMeshLayer::hasSimplifiedMeshes() const
+{
+  return ( mTriangularMeshes.size() > 1 );
```

Comment:
SHould this be > 0? If it's correct, a comment would be useful to explain why

Reply:
you're right...

Reply:
isn't mTriangularMeshes[0] base triangular mesh? so if mTriangularMeshes == 0, it has not base trianguler mesh, nor simplified meshed, if mTriangularMeshes == 1, it has base trinagular mesh, and if mTriangularMeshes > 1 it has simplified meshes & base mesh

Reply:
@PeterPetrik exactly, I was a bit confusing last evening, due to a long day... As unit test made me back to the good road, only a comment is added...

Reply:
testing, such an underestimated power